src/core/handle.ml

```diff
@@ -245,7 +259,61 @@ let handle_cmd : sig_state -> p_command -> sig_state * proof_data option =
       (* Actually add the symbol to the signature. *)
       out 3 "(symb) %s ≔ %a\n" x.elt pp_term t;
       let d = if op then None else Some(t) in
-      let ss = Sig_state.add_symbol ss expo Defin Eager x a impl d in
+      (fst (Sig_state.add_symbol ss expo Defin Eager x a impl d), None)
+  | P_inductive(ms, id, a, l)     ->
+      (* Verify the modifiers. *)
+      let (prop, e, mstrat) = handle_modifiers ms in
+      if prop <> Defin then
+        fatal cmd.pos "Property modifiers cannot be used in inductive types.";
+      if mstrat <> Eager then
+        fatal cmd.pos "Pattern matching strategy modifiers cannot be used \
+                       in inductive types.";
+      (* Add the inductive type in the signature *)
+      let (ss, sym_typ) = handle_symbol ss e Injec Eager id [] a in
```

Comment:
rename sym_typ into sind or sym_ind to not confuse the symbol with a type

Reply:
better: ind_sym as you use ind_ as prefix for other variables (ind_typ and ind_name)

Reply:
I propose the following naming convention:
- ind_ prefix for things related to the inductive type: ind_sym
- rec_ prefix for things related to the induction principle: rec_typ, rec_name, rec_sym

Reply:
This requires the following renamings: sym_typ -> ind_sym, ind_typ -> rec_typ, ind_name -> rec_name, sym_ind -> rec_sym.

Reply:
and sym_cons -> cons_sym because there is cons_list too.

Reply:
and principle -> ind_principle?

Reply:
gen_rec_typ and gen_rec_rules
change_ind_typ_codom -> gen_ind_typ_codom as well
(gen for generate)